test/e2e/suite/scenario/crl.go

```diff
@@ -0,0 +1,316 @@
+
+			utils.LeafCRLUnrevoke()
+			utils.IntermediateCRLRevoke()
+
+			// verify without cache
+			notation.ExpectFailure().Exec("verify", artifact.ReferenceWithDigest(), "-d").
+				MatchErrKeyWords(
+					VerifyFailed,
+					"CRL file cache miss",
+					"Retrieving crl bundle from file cache with key",
+					"Storing crl bundle to file cache with key",
+					"is revoked",
+					"OCSP check failed with unknown error and fallback to CRL check for certificate #2",
+				)
+
+			// verify with cache
+			notation.ExpectFailure().Exec("verify", artifact.ReferenceWithDigest(), "-d").
+				MatchErrKeyWords(
+					VerifyFailed,
+					"Retrieving crl bundle from file cache with key",
+					"is revoked",
+					"OCSP check failed with unknown error and fallback to CRL check for certificate #2",
+				).
+				NoMatchErrKeyWords(
+					"CRL file cache miss",
+					"Storing crl bundle to file cache with key",
+				)
+		})
+	})
+
+	It("successfully completed with cache creation error in debug log", func() {
+		Host(CRLOptions(), func(notation *utils.ExecOpts, artifact *Artifact, vhost *utils.VirtualHost) {
+			notation.Exec("sign", artifact.ReferenceWithDigest()).
+				MatchKeyWords(SignSuccessfully)
+
+			utils.LeafCRLUnrevoke()
+			utils.IntermediateCRLUnrevoke()
+
+			if err := os.MkdirAll(vhost.AbsolutePath(".cache"), 0500); err != nil {
+				Fail(err.Error())
+			}
+			defer os.Chmod(vhost.AbsolutePath(".cache"), 0700)
+
+			// verify without cache
+			notation.Exec("verify", artifact.ReferenceWithDigest(), "-d").
+				MatchKeyWords(
+					VerifySuccessfully,
+					"Warning: failed to create crl file cache",
+				).
+				MatchErrKeyWords(
+					"OCSP check failed with unknown error and fallback to CRL check for certificate #2",
+					`"GET" "http://localhost:10086/intermediate.crl"`,
+					`"GET" "http://localhost:10086/leaf.crl"`,
+				).
+				NoMatchErrKeyWords(
+					"is revoked",
+				)
+		})
+	})
+
+	It("failed with revoked leaf certificate and cache creation error in debug log", func() {
+		Host(CRLOptions(), func(notation *utils.ExecOpts, artifact *Artifact, vhost *utils.VirtualHost) {
+			notation.Exec("sign", artifact.ReferenceWithDigest()).
+				MatchKeyWords(SignSuccessfully)
+
+			utils.LeafCRLRevoke()
+			utils.IntermediateCRLUnrevoke()
+
+			if err := os.MkdirAll(vhost.AbsolutePath(".cache"), 0500); err != nil {
+				Fail(err.Error())
+			}
+			defer os.Chmod(vhost.AbsolutePath(".cache"), 0700)
+
+			// verify without cache
+			notation.ExpectFailure().Exec("verify", artifact.ReferenceWithDigest(), "-d").
+				MatchKeyWords(
+					"Warning: failed to create crl file cache",
+				).
+				MatchErrKeyWords(
+					VerifyFailed,
+					"OCSP check failed with unknown error and fallback to CRL check for certificate #2",
+					`"GET" "http://localhost:10086/intermediate.crl"`,
+					`"GET" "http://localhost:10086/leaf.crl"`,
+					"is revoked",
+				)
+		})
+	})
+
+	It("successfully completed with cache read error in debug log", func() {
```

Comment:
I see we have `cache creation` and `cache read` error cases, how about `cache set` error case?

Reply:
Yes, the `cache read` test case also includes `cache set` error. Updated the test name.